src/util/hooks.js

```diff
@@ -0,0 +1,19 @@
+/* @flow */
+
+export function callThroughHook<T>(type: string, fn: () => T): T {
+  if (typeof global === 'undefined') {
```

Comment:
Maybe we should move these checks outside of the function call and make them static? Also, instead of a callback interface, shouldn't we promote a `Promise` based interface? That would encourage `async` and `await` usage and would also simplify the code:
```js
const hooks = typeof global !== 'undefined' && global.yarnHooks || {};
const defaultHook = () => Promise.resolve();

export function triggerHook<T>(type): Promise<T> {
    return (hooks[type] || defaultHook)();
};
```

Reply:
We can't, otherwise the hooks will have no way to know when the actions start and end (unless I create one extra hook for each action to mark the end).